README.md

```diff
@@ -1,2 +1,54 @@
 # golang-ci
-CircleCI orb for performing common tasks from the CI of a project written in Go
+CircleCI orb for performing common CI tasks for a Go project
+
+#### How to use this project
+Please refer to [CircleCI registry](https://circleci.com/orbs/registry/orb/financial-times/golang-ci) for information how to use this orb.
+
+#### How the CI steps for this orb work
+
+There are two CircleCI workflows associated with this project. They have different behaviour depending on whether they are triggered by commit push or by creating/pushing specific git tag.
+1. On commit
+
+When you are pushing a commit in any branch including master, the first workflow is going to: 
+* lint the orb's code
+* packed it into single yaml file 
+* deploy it as "dev:alpha" version. These dev versions of the orb are not visible in the CircleCI registry but can be referred by other projects and can be tested.
+* trigger a new CircleCI pipeline. The new pipeline actually allows the just uploaded dev version of the orb to be used. There is no way to refer to the newly deployed dev version of the orb in the same pipeline that it's been deployed.
+
+When you are pushing a commit, the second workflow is going to:
+* run integration tests for the orb using the just deployed dev version. The integration tests in our cases are actually running the main orb's jobs and ensuring they succeed.
+
+2. On tag
+
+When you are creating a new tag from master branch (you can use the simple GitHub UI to create new release with associated git tag), the first workflow is going to:
+* lint the orb's code
+* packed it into single yaml file 
+* deploy it as "dev:alpha" version
```

Comment:
So we can have a single testing version of the orb at any given time?

Reply:
Only one dev versions is deployed by the pipeline. If you are interested in deploying different versions with different names, you can do it manually.